docs/DeveloperGuide.adoc

```diff
@@ -621,6 +620,31 @@ image::HideInfoSequenceDiagrams.png[]
 
 The censored data are only hidden from the window, not changed in the database. The `BrowserPanel` on the bottom right of the window will still show all the information including the hidden information.
 
+=== [Proposed] Improved `delete` functionality
+
+==== Current Implementation
+When the user deletes a loan from the LoanBook, it deletes the correct loan, then commits the result into memory. What this entails is that the entire XML file gets overwritten when each delete command is being called.
+
+==== Proposed implementation:
+From above, there is already a `LoanStatus` that has been implemented. What this does is that it allows the loan to be marked as deleted by setting the enum to the appropriate value. This would result in the LoanBook not needing to recommit every change.
+
+From here, when the user decides to close the LoanBook, it will then loop through all the loans and filter out the ones which have no been delete. These are the loans that would be kept and saved in the LoanBook.
+
+==== Design Considerations
+**Aspect: Deleting a loan**
+
+* **Alternative 1 (current choice)**: we remove each loan as they are deleted
+** Pros: Easy to implement
+** Cons: Large time taken for memory operations as reading and writing to the data is required.
+
+* Alternative 2 (proposed implementation): Using an enum to implement lazy deletion.
+** Pros: Much more time efficient due to lazy deletion
+** Cons: Implementation is harder, and changing to this implementation would affect the undo and redo features.
+
+* Alternative 3: Fix the LoanBook storage manager so that it does not rewrite the entire XML file after each delete operation.
+** Pros: All operations are now lazy. Much better performance of the application because commit operations do not need to change the large XML file.
+** Cons: Large refactoring work required. May potentially affect the timeline of development of the other functions.
```

Comment:
Inconsistent tense in the titles for the alternatives. Try:
1) Remove each loan as they are deleted.
2) Use an enum to implement lazy deletion.
(3 is fine.)